classes/CartRule.php

```diff
@@ -295,6 +295,34 @@ public static function getIdByCode($code)
     }
 
     /**
+     * Check if some cart rules exists today for the given customer
+     *
+     * @param int $idCustomer
+     *
+     * @return bool
+     */
+    public static function haveCartRuleToday($idCustomer)
+    {
```

Comment:
in this method you only check if cart rule starts or ends today
not if it is valid today
Is there a reason for this restriction ?

Reply:
I wanted to use the most generic query to make it easily cacheable